add playbook for karrot.world and fix behavior on bare debian

tiltec opened the pull request:
I copied it from the foodsharing taiwan playbook, adjusted secrets and added the backup script

I didn't run it yet, was afraid of having to fix a broken production deployment.

The site user needs access to the configured offsite location, for now by manually copying the private key from the deploy user. I wonder if I should add that private key to the vault?

The site user also needs access to progres to create arbitrary databases and delete them later on. It would probably be better to have only one backup test db. Maybe this can also be added to the playbook...

tiltec committed:
add playbook for karrot.world

tiltec committed:
remove db section

nicksellen commented:
> I didn't run it yet, was afraid of having to fix a broken production deployment.

We could co-ordinate to be online at the same time, at a reasonably out-of-hours type time, and fix any crap together. Or wait till we're together in person.

> The site user needs access to the configured offsite location, for now by manually copying the private key from the deploy user. I wonder if I should add that private key to the vault?

The ansible user module can generate an ssh key ---> http://docs.ansible.com/ansible/latest/user_module.html

So we could use that, the rsync to offsite location would fail until someone added the public key to it, but thats seems fine. It's a oneoff.

> The site user also needs access to progres to create arbitrary databases and delete them later on. It would probably be better to have only one backup test db. Maybe this can also be added to the playbook...

Would seem nice just to grant access to the site user (or even a seperate site-specific backup user) for just that backup database. Easy to automate it all with ansible :)

tiltec committed:
Merge branch 'master' into add-karrot-world

tiltec committed:
add ssh key

tiltec commented:
Right now, the playbook depends on a GPG key to exist to encrypt the backups. I was looking for ways to manage this:
- perhaps pregenerate the key and store it in vault, then import to system
- use https://github.com/juju4/ansible-gpgkey_generate or similar to generate a key on the fly

I like the first option better, then all people who know the vault password can also decrypt the backups.

nicksellen commented:
Yup, 1 sounds good to me.

tiltec committed:
remove the backup checking stuff

I find it too complex to think about postgres roles and authorization

tiltec committed:
add gpg secret key

I didn't try it and it seems a bit ugly to put it directly in the playbook. But if I put it in the secrets file, I would have to encode it some (gpg secret key is binary)

tiltec commented on playbooks/karrot-world/setup.playbook.yml:
```diff
@@ -0,0 +1,293 @@
+        postgresql_database: fstool
+        gpg_id: B88A03E5
+        offsite_server: tiltec@81.30.158.213:backup/
+        gpg_secret_key: !vault |
```
I couldn't think of a better way than to include the secret key inline (also not sure if it works).

tiltec committed:
add vagrant to test setup

tiltec committed:
improve server setup from bare debian

tiltec committed:
make karrot-world work on vagrant
